tests/run-make/no-input-file/rmake.rs

```diff
@@ -0,0 +1,9 @@
+extern crate run_make_support;
+
+use run_make_support::{diff, rustc};
+
+fn main() {
+    let output = rustc().arg("--print").arg("crate-name").run_fail_assert_exit_code(1);
```

Comment:
One last nit: can you add a `print` helper method to `Rustc`? So the callsite becomes something like

```rs
rustc().print("crate-name")
```

This PR looks good to me otherwise.